File: 1-Text-Based/C#/Pemilu2024.cs
using System;

class Pemilu2024
{
    private static readonly string[] paslonPemilu2024 = { "Anies-Muhaimin", "Prabowo-Gibran", "Ganjar-Mahfud" };
    private static readonly string[] parpolPemilu2024 = { "PKB", "Gerindra", "PDIP", "Golkar", "Nasdem", "Buruh", "Gelora", "PKS", "PKN", "Hanura", "Garuda", "PAN", "PBB", "Demokrat", "PSI", "Perindo", "PPP", "Ummat" };

    static void Main(string[] args)
    {
        while (true)
        {
            Console.WriteLine("\n======================================");
            Console.WriteLine("     PROGRAM PEMILIHAN UMUM 2024     ");
            Console.WriteLine("--------------------------------------");
            Console.WriteLine("1. Tampilkan Nama Paslon dan Partai");
            Console.WriteLine("2. Pilih Paslon dan Partai");
            Console.WriteLine("======================================");

            Console.Write("Masukkan pilihan Anda : ");
            int choice = int.Parse(Console.ReadLine());

            if (choice == 1)
            {
                Console.WriteLine("\n======================================");
                Console.WriteLine("Pasangan Calon :");
                for (int i = 0; i < paslonPemilu2024.Length; i++)
                {
                    Console.WriteLine((i + 1) + ". " + paslonPemilu2024[i]);
                }

                Console.WriteLine("\nPartai Politik :");
                for (int i = 0; i < parpolPemilu2024.Length; i++)
                {
                    Console.WriteLine((i + 1) + ". " + parpolPemilu2024[i]);
                }
                Console.WriteLine("======================================");
            }
            else if (choice == 2)
            {
                Console.Write("\nMasukkan Nomor Paslon yang dipilih : ");
                int paslon_pilihan = int.Parse(Console.ReadLine());
                if (!(1 <= paslon_pilihan && paslon_pilihan <= paslonPemilu2024.Length))
                {
                    Console.WriteLine("Nomor paslon tidak valid.");
                    continue;
                }

                string[] partai_koalisi;
                if (paslonPemilu2024[paslon_pilihan - 1].Equals("Anies-Muhaimin"))
                {
                    partai_koalisi = new string[] { "PKB", "Nasdem", "PKS", "Ummat" };
                }
                else if (paslonPemilu2024[paslon_pilihan - 1].Equals("Ganjar-Mahfud"))
                {
                    partai_koalisi = new string[] { "PDIP", "Hanura", "Perindo", "PPP" };
                }
                else if (paslonPemilu2024[paslon_pilihan - 1].Equals("Prabowo-Gibran"))
                {
                    partai_koalisi = new string[] { "Gerindra", "Golkar", "Buruh", "Gelora", "PKN", "Garuda", "PAN", "PBB", "Demokrat", "PSI" };
                }
                else
                {
                    partai_koalisi = parpolPemilu2024;
                }

                Console.WriteLine("\n======================================");
                Console.WriteLine("Partai Politik Koalisi " + paslonPemilu2024[paslon_pilihan - 1] + " :");
                for (int i = 0; i < partai_koalisi.Length; i++)
                {
                    Console.WriteLine((i + 1) + ". " + partai_koalisi[i]);
                }
                Console.Write("======================================");

                Console.Write("\nMasukkan Nomor Partai yang dipilih : ");
                int partai_pilihan = int.Parse(Console.ReadLine());
                if (!(1 <= partai_pilihan && partai_pilihan <= partai_koalisi.Length))
                {
                    Console.WriteLine("Nomor partai tidak valid.");
                    continue;
                }

                Console.WriteLine("\n======================================");
                Console.WriteLine("             PILIHAN ANDA             ");
                Console.WriteLine("--------------------------------------");
                Console.WriteLine("Pasangan Calon : " + paslonPemilu2024[paslon_pilihan - 1]);
                Console.WriteLine("Partai Politik : " + partai_koalisi[partai_pilihan - 1]);
                Console.WriteLine("======================================");
            }
            else
            {
                Console.WriteLine("Pilihan tidak valid. Silakan masukkan 1 atau 2.");
                continue;
            }

            Console.Write("\nApakah Anda ingin keluar? (y/n): ");
            string exit_choice = Console.ReadLine().ToLower();
            if (exit_choice.Equals("y"))
            {
                break;
            }
        }

        Console.WriteLine("\n");
    }
}

File: 1-Text-Based/Go/pemilu-2024.go
package main

import (
    "fmt"
)

var paslonPemilu2024 = []string{"Anies-Muhaimin", "Prabowo-Gibran", "Ganjar-Mahfud"}
var parpolPemilu2024 = []string{"PKB", "Gerindra", "PDIP", "Golkar", "Nasdem", "Buruh", "Gelora", "PKS", "PKN", "Hanura", "Garuda", "PAN", "PBB", "Demokrat", "PSI", "Perindo", "PPP", "Ummat"}

func main() {
    for {
        fmt.Println("\n======================================")
        fmt.Println("     PROGRAM PEMILIHAN UMUM 2024     ")
        fmt.Println("--------------------------------------")
        fmt.Println("1. Tampilkan Nama Paslon dan Partai")
        fmt.Println("2. Pilih Paslon dan Partai")
        fmt.Println("======================================")

        var choice int
        fmt.Print("Masukkan pilihan Anda : ")
        fmt.Scanln(&choice)

        if choice == 1 {
            fmt.Println("\n======================================")
            fmt.Println("Pasangan Calon :")
            for i, paslon := range paslonPemilu2024 {
                fmt.Printf("%d. %s\n", i+1, paslon)
            }

            fmt.Println("\nPartai Politik :")
            for i, partai := range parpolPemilu2024 {
                fmt.Printf("%d. %s\n", i+1, partai)
            }
            fmt.Println("======================================")
        } else if choice == 2 {
            var paslonPilihan int
            fmt.Print("\nMasukkan Nomor Paslon yang dipilih : ")
            fmt.Scanln(&paslonPilihan)
            if !(1 <= paslonPilihan && paslonPilihan <= len(paslonPemilu2024)) {
                fmt.Println("Nomor paslon tidak valid.")
                continue
            }

            var partaiKoalisi []string
            switch paslonPemilu2024[paslonPilihan-1] {
            case "Anies-Muhaimin":
                partaiKoalisi = []string{"PKB", "Nasdem", "PKS", "Ummat"}
            case "Ganjar-Mahfud":
                partaiKoalisi = []string{"PDIP", "Hanura", "Perindo", "PPP"}
            case "Prabowo-Gibran":
                partaiKoalisi = []string{"Gerindra", "Golkar", "Buruh", "Gelora", "PKN", "Garuda", "PAN", "PBB", "Demokrat", "PSI"}
            default:
                partaiKoalisi = parpolPemilu2024
            }

            fmt.Println("\n======================================")
            fmt.Printf("Partai Politik Koalisi %s :\n", paslonPemilu2024[paslonPilihan-1])
            for i, partai := range partaiKoalisi {
                fmt.Printf("%d. %s\n", i+1, partai)
            }
            fmt.Println("======================================")

            var partaiPilihan int
            fmt.Print("\nMasukkan Nomor Partai yang dipilih : ")
            fmt.Scanln(&partaiPilihan)
            if !(1 <= partaiPilihan && partaiPilihan <= len(partaiKoalisi)) {
                fmt.Println("Nomor partai tidak valid.")
                continue
            }

            fmt.Println("\n======================================")
            fmt.Println("             PILIHAN ANDA             ")
            fmt.Println("--------------------------------------")
            fmt.Printf("Pasangan Calon : %s\n", paslonPemilu2024[paslonPilihan-1])
            fmt.Printf("Partai Politik : %s\n", partaiKoalisi[partaiPilihan-1])
            fmt.Println("======================================")
        } else {
            fmt.Println("Pilihan tidak valid. Silakan masukkan 1 atau 2.")
            continue
        }

        var exitChoice string
        fmt.Print("\nApakah Anda ingin keluar? (y/n): ")
        fmt.Scanln(&exitChoice)
        if exitChoice == "y" {
            break
        }
    }
    fmt.Println("\n")
}
